### PR TITLE
WebDev Team Modal

### LINKED ISSUE #98: Create Web Team Modal
### Description
Create Modal box for Web team

### Screenshots/testimonials

I'll update it with an example

### LINKED ISSUE #141: Create design for WebDev team modal
### Description

With reference to #98, provide a design for WebDev team modal to be displayed on the [hG website](https://houseofgeeks.netlify.app/). You should be following space theme for this. Kindly go through the website to get the design asthetics of the website and make this relevant to the present design. The modal should contain all the avatar/card of all the contributors of the project. You should also make a design for mobile version of the modal.

### Screenshots/testimonials
NA

### PR DESCRIPTION
**Pull request by tusharjain0022**
Fixes #98 
Fixes #141  
<Do not delete this template>

Description: <provide a description> I've created webdev team modal which will be triggered on clicking `webdev team House of Geeks` in the footer.

#### Checklist
- [x] I've followed the Contributing guidelines provided in the repository.
- [x] I've made the changes which were demanded in the linked issue.
- [x] I've tested my code on a Chromium based browser.
- [ ] I've tested my code on Mozilla Firefox.
- [x] My code gave a clean console on debugging. (no warnings/errors)

#### Testimonial
<provide a relevant screenshot or any testimonial for the solved issue if applicable>

Screenshot (24)

### CODE REVIEW

File: client/src/components/Footer/showName.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            rel="noopener noreferrer"
+            href={"https://twitter.com/" + userInfo.twitter_username}
+          >
+            <Twitter className="icon"></Twitter>
+          </a>
+
+          <a
+            target="_blank"
+            rel="noopener noreferrer"
+            href={"mailto:" + userInfo.mail}
```
Comment (EmperorYP7) [inline client/src/components/Footer/showName.js]: This field isn't working

File: client/src/components/Footer/modal.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+function Modal() {
+  const [userData, setUserData] = useState([]);
+
+  useEffect(() => {
+    getGitHubUserWithFetch();
+  }, []);
+
+  const getGitHubUserWithFetch = async () => {
+    const response = await fetch(
```
Comment (EmperorYP7) [inline client/src/components/Footer/modal.js:14]: How about using axios and promises to handle the fetch?

File: client/src/components/Footer/Footer.js [hunk 3]
```diff
@@ -1,60 +1,94 @@
...
+              href="https://www.linkedin.com/company/hgiiitranchi/"
+            >
+              <LinkedIn className="icons fa-7x"></LinkedIn>
+            </a>
+          </div>
+        </div>
+        <div className="row">
+          <div className="col-12 col-md-4 col-sm-4 text-center text-white">
+            <p data-toggle="modal" data-target="#exampleModal">
+              WebDev Team House Of Geeks
```
Comment (EmperorYP7) [inline client/src/components/Footer/Footer.js:76]: Lets have some hover effect on this to make its action more obvious to the user.

File: client/src/components/Footer/showName.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+function ShowName({ prop }) {
+  const [userInfo, setUserInfo] = useState({});
+
+  useEffect(() => {
+    getGitHubUserWithFetch();
+  });
+
+  const getGitHubUserWithFetch = async () => {
+    const response = await fetch(prop);
```
Comment (EmperorYP7) [inline client/src/components/Footer/showName.js:17]: use promises here as well.

**Review by EmperorYP7 — CHANGES_REQUESTED** [review]
Nicely implemented @tusharjain0022! 🤩 Some minor updates and we're good to go! :rocket:

File: client/src/components/Footer/modal.js (hunk 2 above)
Comment (anktrj) [reply client/src/components/Footer/modal.js:14]: I don't know.
For now, it's good.

File: client/src/components/Footer/showName.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import { ReactComponent as Github } from "./img/github-brands.svg";
+import { ReactComponent as Twitter } from "./img/twitter-brands.svg";
+import { ReactComponent as Mail } from "./img/envelope-solid.svg";
+
+function ShowName({ prop }) {
+  const [userInfo, setUserInfo] = useState({});
+
+  useEffect(() => {
+    getGitHubUserWithFetch();
+  });
```
Comment (anktrj) [inline client/src/components/Footer/showName.js]: fix this infinite call is happening!
pass an empty array to the useEffect hook

**Review by ankiiitraj — CHANGES_REQUESTED** [review]

**Comment by ankiiitraj** [comment]
Also don't render those social icons, links to which are not available!

File: client/src/components/Footer/Footer.js (hunk 3 above)
Comment (tusharjain0022) [reply client/src/components/Footer/Footer.js:76]: Issue resolved

File: client/src/components/Footer/showName.js (hunk 1 above)
Comment (tusharjain0022) [reply client/src/components/Footer/showName.js]: Issue resolved

File: client/src/components/Footer/showName.js (hunk 5 above)
Comment (tusharjain0022) [reply client/src/components/Footer/showName.js]: Issue resolved , please take a look at it again.

**Comment by tusharjain0022** [comment]
> Also don't render those social icons, links to which are not available!

Issue resolved .

**Comment by ankiiitraj** [comment]
Does this look good? Just added a border of 3px
Screenshot from 2020-10-26 15-31-27

**Review by ankiiitraj — APPROVED** [review]
Cool

**Comment by tusharjain0022** [comment]
> Does this look good? Just added a border of 3px
> Screenshot from 2020-10-26 15-31-27

Looks nice, I'll add this in my next commit and will fix the horizontal overflow of image-hover text.

**Comment by hariom1625** [comment]
Nice work @tusharjain0022 :-)

Suggestion: I think the width of the modal can be reduced a bit or adding a slight margin along the sides, at present, it is covering most of the screen shadowing the overlay feature experience, and also profile images are appearing quite large.

Suggestions @tusharjain0022 @ankiiitraj @EmperorYP7 ?

**Comment by ankiiitraj** [comment]
The `WebDev Team House Of Geeks` don't seem intutively a click thing. I'll open an issue for it!
This PR LGTM. Ready to be merged after hariom and yash review!

**Review by ankiiitraj — APPROVED** [review]

**Comment by tusharjain0022** [comment]
> Nice work @tusharjain0022 :-)
> 
> Suggestion: I think the width of the modal can be reduced a bit or adding a slight margin along the sides, at present, it is covering most of the screen shadowing the overlay feature experience, and also profile images are appearing quite large.
> 
> Suggestions @tusharjain0022 @ankiiitraj @EmperorYP7 ?

View with decreased width of modal, looks good to me. Should I commit this change @hariom1625 @EmperorYP7 ?
Screenshot (27)

**Comment by EmperorYP7** [comment]
> View with decreased width of modal, looks good to me. Should I commit this change @hariom1625 @EmperorYP7 ?

Go ahead @tusharjain0022. Afterall,  you can always revert those changes. :smile:

**Comment by tusharjain0022** [comment]
Modal width decreased !

**Review by hariom1625 — APPROVED** [review]

**Pull request merged**